backend/tests/building.spec.js

```diff
@@ -0,0 +1,84 @@
+/*
+* @Author: Milan Donhowe
+* @Date:   4/23/2021
+* @Last Modified By:  Milan Donhowe
+* @Last Modified Time:  4/23/2021
+* @Copyright:  Oregon State University 2021
+* @Description: Unit tests for building.js, "allbuildings" endpoint
+*/
+
+/* CORS testing imports */
+const testConfig = require('./assertedData/test_config.json')
+const CORSUtil = require('./utility/cors_test_utility.js')
+const server = testConfig['serverOrigin']
+const client = testConfig['clientOrigin']
+
+const MOCK_REQUEST_EVENT = {
+    headers: {
+        origin: `${client.scheme}://${client.host}`
+    }
+}
+
+const csvHelper = require('./utility/csv_to_obj.js')
+let mockSQLResponse = csvHelper.csv2object('./tests/assertedData/sql_mock_allbuildings.csv')
+
+
+/* Lambda Common Layer mocks */
+const mockResponse = require(`${testConfig.so_namespace}/response.js`)
+jest.mock(
+    '/opt/nodejs/response.js',
+    () => { return mockResponse },
+    {virtual: true}
+
+)
+
+/* Mock MySQL Database */
+jest.mock(
+    '/opt/nodejs/sql-access.js',
+    () => {return {
+        // claim db is up
+        connect: async () => { return Promise.resolve() },
+        // return mock sql data
+        query: async () => { 
+            return Promise.resolve(mockSQLResponse)
+        }
+    }},
+    {virtual: true}
+)
+
+// Mock dashboard specific dependencies
+const modelMocks = ['meter_classes.js', 'models/meter.js', 'models/meter_group.js', 'models/building.js']
+for (modelMock of modelMocks){
+    const mock = require(`../dependencies/nodejs/${modelMock}`)
+    jest.mock(
+        `/opt/nodejs/${modelMock}`,
+        () => {return mock},
+        {virtual: true}
+    ) 
+}
+ 
+// stub "User" module (not currently used by endpoint)
+jest.mock('/opt/nodejs/user.js', () => {null}, {virtual:true})
+
+const AllBuildings = require('../app/building.js')
+
+describe('Testing allbuildings endpoint...', () => {
```

Comment:
Is there a benefit to having a separate file for each endpoint? I feel that we can eliminate code duplication by running tests against allBuildings, campaigns, etc in one file (and use a smarter mock that grabs the correct data based on the request url).

Reply:
Perhaps we can group these somehow- maybe by the VueX module they correspond to? What are your thoughts @MilanDonhowe

Reply:
Yeah, there's not a huge benefit in keeping them separated beyond just making the database mocks easier to write.  We could try grouping them based on their corresponding VueX module, my only concern would be that in the event that we modified or drastically re-structured the VueX store (which probably won't happen, but is worth mentioning) it would be a bit of a hassle to re-structure the backend tests.  For now, I'll go ahead and group by VueX modules and see how it works.

I'll also eliminate a lot of duplicate code by adding a [setup script ](https://jestjs.io/docs/configuration#setupfiles-array) to `require()` a lot of redundant modules.